docs/guides/darwin.md

```diff
@@ -0,0 +1,202 @@
+    a profile on it (instructions below).
+-   An Apple TV or HomePod running at least tvOS 15.6. You will need to install
+    a profile on it (instructions below). If you have multiple Apple TVs and/or
+    HomePods, you will need to install the profile on all of them.
+-   The iPhone or iPad with the Matter profile must belong to the same iCloud
+    account as the Apple TV or HomePod with the Matter profile.
+-   A test iCloud account is highly recommended for Matter development to avoid
+    interference with existing homes.
+
+## Source Compatibility
+
+Each developer preview release is compatible with a certain SHA from this
+repository.
+
+-   iOS/iPadOS/tvOS 15.6 Developer Preview:
+    [`cfc35951be66a664a6efdadea56d1b8ea6e63e96`](https://github.com/project-chip/connectedhomeip/commits/cfc35951be66a664a6efdadea56d1b8ea6e63e96)
+-   iOS/iPadOS/tvOS 16.0 Developer Preview:
+    [`cfc35951be66a664a6efdadea56d1b8ea6e63e96`](https://github.com/project-chip/connectedhomeip/commits/cfc35951be66a664a6efdadea56d1b8ea6e63e96)
+
+## Profile Installation
+
+### Install the Matter Profile on your iPhone or iPad
+
+Note: The profile will expire automatically after some time
+
+1. Download the
+   [profile](https://developer.apple.com/services-account/download?path=/iOS/iOS_Logs/EnableMatter.mobileconfig)
+   and install it on the iOS/iPadOS 15.6 device. If necessary, email the profile
+   or use AirDrop to transfer the profile to the iOS/iPadOS 15.6 device.
+2. Restart the device.
+
+### Install the Matter Profile on your Apple TV or HomePod
+
+Note: The profile will expire automatically after some time
+
+1. Download the
+   [profile](https://developer.apple.com/services-account/download?path=/iOS/iOS_Logs/EnableMatter.mobileconfig)
+   and install it on the the Apple TV or HomePod running tvOS 15.6. If
+   necessary, use
+   [Apple Configurator 2](https://support.apple.com/guide/apple-configurator-2/connect-devices-to-your-mac-cad9d4b2211e/2.14/mac/11.3.1)
+   to transfer the profile to the Apple TV or HomePod.
+2. Restart the device.
+
+### Ensuring Your Matter Accessory Works with iOS/iPadOS/tvOS
+
+1. Clone the [Matter repo](https://github.com/project-chip/connectedhomeip.git)
+2. Checkout the specific commit hash (from [above](#source-compatibility)) for
+   maximum compatibility with your installed release:
+    - Example command for SHA `cfc35951be66a664a6efdadea56d1b8ea6e63e96`:
+      `$ git checkout cfc35951be66a664a6efdadea56d1b8ea6e63e96`
+
+In order to work with iOS/iPadOS/tvOS 15.6 or greater, device types as defined
+in the Matter Device Library spec are used to determine accessory categories.
+Ensure the right device type is set for each endpoint.
+
+-   For the `all-clusters-app` as an example, this can be set in
+    `FIXED_DEVICE_TYPES` in `endpoint_config.h`
+
+-   A light on Endpoint 1 can be set in `FIXED_DEVICE_TYPES` in
+    `endpoint_config.h` for `all-clusters-app` like so:
```

Comment:
Actually, you have to set FIXED_DEVICE_TYPES, FIXED_DEVICE_TYPE_OFFSETS, and FIXED_DEVICE_TYPE_LENGTHS.

In particular, the "1" in the array below is not the endpoint id; it's the device revision number.